src/middlewared/middlewared/utils/__init__.py

```diff
@@ -320,7 +321,10 @@ def filterop(f):
             rv = sorted(rv, key=lambda x: x[o], reverse=reverse)
 
     if options.get('get') is True:
-        return rv[0]
+        try:
+            return rv[0]
+        except IndexError:
+            raise MatchNotFound()
```

Comment:
filter_utils is not the only place it can raise IndexError, it can also happen in datastore.query